### DISCUSSION TITLE
Support for method pointers

### DISCUSSION THREAD

**Discussion by radare** [vlang/v#11659] [Ideas]
**V version:** V 0.1.25 c12985d
**OS:** macOS 10.15

**What did you do?**

ideally it will be nice to be able to hold in a variable the instance and the function pointer associated with the given statement, like in a closure

```
fn res(s string) {
	println(s)
}

struct Foo {i int}

fn new_foo() &Foo {
	return &Foo{0}
}

fn (a &Foo)fai() {
	println(a)
}

fn main() {
        // works
	c := res
	c('foo')

	d := new_foo()
	d.fai()
	e := d.fai //  fails
	e()
}
```

**What did you expect to see?**
```
foo
0x6020000000d0
```
 
**What did you see instead?**

```
$ v run test_method_pointer.v
next token = `(`
test_method_pointer.v:23:3: syntax error: unexpected `e`, expecting `(`
   21| 	d.fai()
   22| 	e := d.fai
   23| 	e()
         ^
   24| }
```

**Comment by Ivo-Balbaert** [comment]
The V errors are now (v 0.1.27):

test.v:22:9: error: unknown field `Foo.fai` 
   20 |     d := new_foo()
   21 |     d.fai()
   22 |     e := d.fai //  fails
      |            ~~~
   23 |     e()
   24 | }
test.v:23:2: error: unknown function: e 
   21 |     d.fai()
   22 |     e := d.fai //  fails
   23 |     e()
      |     ~~~
   24 | }
